torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -731,6 +745,56 @@ TensorView* clamp(TensorView* in, Val* min_val, Val* max_val) {
   return clamp(in->as<Val>(), min_val, max_val)->as<TensorView>();
 }
 
+// sum_to operator
+
+inline bool shouldReduceSumTo(const int shape, const Val* root_id_extent) {
+  return shape == 1 && !root_id_extent->isOneInt();
+}
+
+inline bool shouldReduceSumTo(const Val* shape, const Val* root_id_extent) {
+  return shape->isOneInt() && !root_id_extent->isOneInt();
+}
+
+// Name of sum_to is different from NV fuser naming,
+// this is to align with the operator name
+template <typename T>
+TensorView* sum_to_impl(TensorView* v1, const std::vector<T>& shape) {
```

Comment:
I appreciate the generic implementation here, but I think we should explicitly take in a `std::vector<Int*>` and only a `std::vector<Int*>`.

Reply:
I see. Refactored the code slightly to remove std::vector<int> support and now only has std::vector<Int*> explicitly.